ethers-providers/src/lib.rs

```diff
@@ -421,6 +424,15 @@ pub trait Middleware: Sync + Send + Debug {
         self.inner().get_logs(filter).await.map_err(FromErr::from)
     }
 
+    /// Returns a stream of logs are loaded in pages of given page size
+    fn get_logs_paginated<'a>(
+        &'a self,
+        filter: &Filter,
+        page_size: u64,
+    ) -> LogQuery<'a, Self::Provider> {
+        self.inner().get_logs_paginated(filter, page_size)
+    }
```

Comment:
@prestwich wdyt about unifying this logic in the default get_logs behavior?

Reply:
i think it's a pretty reasonable way to smooth over inconsistent RPC paging requirements 🤔